File: frontend/src/services/technicianService.ts
import api from "@/lib/api";
import type { Technician } from "@/types/technician";

export const technicianService = {
  // ============================================================
  // GET ALL TECHNICIANS
  // GET /api/technicians
  // ============================================================

  getAllTechnicians: async (): Promise<Technician[]> => {
    const response = await api.get<Technician[]>(
      "/technicians"
    );

    return response.data;
  },

  // ============================================================
  // GET AVAILABLE TECHNICIANS
  // GET /api/technicians/available
  // ============================================================

  getAvailableTechnicians: async (): Promise<Technician[]> => {
    const response = await api.get<Technician[]>(
      "/technicians/available"
    );

    return response.data;
  },

  // ============================================================
  // GET TECHNICIAN BY ID
  // GET /api/technicians/{id}
  // ============================================================

  getTechnicianById: async (
    id: number
  ): Promise<Technician> => {
    const response = await api.get<Technician>(
      `/technicians/${id}`
    );

    return response.data;
  },
};

File: backend/src/main/java/com/keystone/service/TechnicianService.java
package com.keystone.service;

import java.util.List;

import com.keystone.dto.TechnicianRequest;
import com.keystone.dto.TechnicianResponse;

public interface TechnicianService {

    // =========================================================
    // CREATE TECHNICIAN
    // =========================================================

    TechnicianResponse addTechnician(
            TechnicianRequest request
    );


    // =========================================================
    // GET ALL TECHNICIANS
    // =========================================================

    List<TechnicianResponse> getAllTechnicians();


    // =========================================================
    // GET AVAILABLE TECHNICIANS
    // =========================================================

    List<TechnicianResponse> getAvailableTechnicians();


    // =========================================================
    // GET TECHNICIAN BY ID
    // =========================================================

    TechnicianResponse getTechnicianById(
            Long id
    );


    // =========================================================
    // UPDATE TECHNICIAN
    // =========================================================

    TechnicianResponse updateTechnician(
            Long id,
            TechnicianRequest request
    );


    // =========================================================
    // DELETE TECHNICIAN
    // =========================================================

    void deleteTechnician(
            Long id
    );


    // =========================================================
    // MY PROFILE
    // =========================================================

    TechnicianResponse getMyProfile(
            String email
    );


    // =========================================================
    // UPDATE MY AVAILABILITY
    // =========================================================

    TechnicianResponse updateMyAvailability(
            String email,
            Boolean available
    );
}